src/main/java/seedu/address/storage/JsonSerializableAddressBook.java

```diff
@@ -80,13 +82,18 @@ private void loadDoctorsAndTheirAssignedPatients(AddressBook addressBook)
             }
             addressBook.addDoctor(doctor);
 
+            Set<Patient> patientsAdded = new HashSet<>();
             for (Patient patient : doctor.getPatients()) {
                 // Do not perform a check for duplicate patients here
                 // as the patients are bound to be stored multiple
                 // times under the different doctors they are assigned to.
+                if (patientsAdded.contains(patient)) {
+                    throw new IllegalValueException(MESSAGE_DUPLICATE_PATIENT);
```

Comment:
@praveenkrishna0512 Required to update test cases to improve coverage on line 91